### PR TITLE
Add the -q/--quiet option to don't print informative messages

### PR DESCRIPTION
**Pull request by fabio-porcedda**

### CODE REVIEW

**Comment by jpakkane** [comment]
What is the use case for this?

**Comment by fabio-porcedda** [comment]
When i build a complex project with many subprojects where the meson project is just only a part a prefer to have the less verbose level possible, because otherwise too much noise is generated, normally too much output isn't good for mee, i will use the verbose output only when i'm debugging some problems.

When i build a meson cross compile project it prints 15 lines, just for the configuring it, instead ninja for building the project it use just one line. So i added this option to have a less verbose output of meson where for my use case should be zero or at most one line.

Many tools have a "quiet" option soo it souldn't a too strange thing to have.

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -1610,7 +1610,7 @@ def func_project(self, node, args, kwargs):
             if not mesonlib.version_compare(cv, pv):
                 raise InterpreterException('Meson version is %s but project requires %s.' % (cv, pv))
         self.build.projects[self.subproject] = args[0]
-        mlog.log('Project name: ', mlog.bold(args[0]), sep='')
+        mlog.info('Projectname: ', mlog.bold(args[0]), sep='')
```
Comment (trhd) [inline mesonbuild/interpreter.py]: Is the space removed on purpose?

Comment (fabio-porcedda) [inline mesonbuild/interpreter.py]: It's an error, I have corrected it.

Thanks for catching it up.

**Comment by jpakkane** [comment]
Thank you for this contribution but ultimately we came to the conclusion that even though these messages do take some space, having them printed in every case is a better UX for most users.

Closing this now. We hope you enjoy using Meson regardless and wish you will submit more MRs in the future.

**Pull request closed without merging**